crates/admin/src/cluster_controller/service.rs

```diff
@@ -305,6 +307,14 @@ impl<T: TransportConnect> Service<T> {
                         warn!("Could not trim the logs. This can lead to increased disk usage: {err}");
                     }
                 }
+                Ok(_) = tail_udpates_watcher.changed() => {
+                    // we clone here to avoid holding the read lock in the watcher
+                    // more than necessary.
+                    // todo(azmy): do we really need to keep the updates inside an Arc?
+                    // the on_logs_tail_updates should return quick enough.
+                    let tail = tail_udpates_watcher.borrow_and_update().clone();
+                    logs_controller.on_logs_tail_updates(&tail);
+                }
```

Comment:
We could move this part and the periodic triggering of the find tails check into the `LogsController::run_async_operations` to make it an implementation detail of the `LogsController.